docs/REST API/rest-api.md

```diff
@@ -1052,85 +1052,272 @@ Modifies the Fleet's configuration with the supplied information.
+| enable                          | boolean  | Whether to enable the end user migration workflow for devices migrating from your old MDM solution. |
+| mode                          | string  | The end user migration workflow mode for devices migrating from your old MDM solution. Options are `"voluntary"` or `"forced"`. |
+| webhook_url                          | string  | The webhook url configured to receive requests to unenroll devices migrating from your old MDM solution. |
+
+<br/>
+
+##### mdm.macos_setup
+
+_Available in Fleet Premium._
+
+`mdm.macos_setup` is an object with the following structure:
+
+| Name                              | Type    | Description   |
+| ---------------------             | ------- | -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| enable_end_user_authentication            | boolean | If set to true, end user authentication will be required during automatic MDM enrollment of new macOS devices. Settings for your IdP provider must also be [configured](https://fleetdm.com/docs/using-fleet/mdm-macos-setup-experience#end-user-authentication-and-eula). |
+
+<br/>
+
+##### mdm.macos_settings
+
+`mdm.macos_settings` is an object with the following structure:
+
+| Name                              | Type    | Description   |
+| ---------------------             | ------- | -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| custom_settings                   | list    | _mdm.macos_settings settings_. macOS hosts that belong to no team, and are members of specified labels will have custom profiles applied. |
+| enable_disk_encryption            | boolean | _mdm.macos_settings settings_. _Available in Fleet Premium._ Hosts that belong to no team and are enrolled into Fleet's MDM will have disk encryption enabled if set to true. |
+
+<br/>
+
+##### mdm.windows_settings
+
+`mdm.windows_settings` is an object with the following structure:
+
+| Name                              | Type    | Description   |
+| ---------------------             | ------- | -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| custom_settings                   | list    | Windows hosts that belong to no team, and are members of specified labels will have custom profiles applied. |
 
-Note that when making changes to the `integrations` object, all integrations must be provided (not just the one being modified). This is because the endpoint will consider missing integrations as deleted.
 
 #### Example
```

Comment:
@rachaelshaw maybe pull the example up above tables? Or, maybe even better, break out separate examples for each object because that's the best practice for patching the config.

I previewed out the changes locally and was looking for the example JSON object to make sense of the objects.

Reply:
@rachaelshaw this is such a good start!!! Thanks for putting in the effort to make this doc more user friendly.

> break out separate examples for each object because that's the best practice for patching the config

+1 to this!